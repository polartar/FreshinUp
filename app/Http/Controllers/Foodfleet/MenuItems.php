<?php

namespace App\Http\Controllers\Foodfleet;
use App\Http\Controllers\Controller;
use App\Models\Foodfleet\MenuItem;
use App\Models\Foodfleet\Store;
use App\Http\Resources\Foodfleet\MenuItem as MenuItemResource;
use Spatie\QueryBuilder\Filter;
use Spatie\QueryBuilder\QueryBuilder;
use Symfony\Component\HttpFoundation\Response as SymfonyResponse;
use Illuminate\Http\Request;

class MenuItems extends Controller
{
    public function index(Request $request, $uuid)
    {
        $items = QueryBuilder::for(MenuItem::class, $request)
            ->allowedIncludes(['store'])
            ->allowedFilters([
                'store',
                Filter::exact('uuid'),
                Filter::exact('store_uuid')
            ]);
        return MenuItemResource::collection($items->get());
    }
}
