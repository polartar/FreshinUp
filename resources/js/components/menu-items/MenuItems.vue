<template>
  <v-card>
    <v-card-title class="grey--text">
      <v-layout
        row
        space-between
        align-center
      >
        <v-flex>
          <h3>Fleet Member Menu</h3>
        </v-flex>
        <v-flex shrink>
          <v-btn
            v-show="!newDialog"
            slot="activator"
            color="primary"
            text
            @click="newDialog = true"
          >
            <v-icon
              dark
              left
            >
              add_circle_outline
            </v-icon>Add new menu item
          </v-btn>
        </v-flex>
      </v-layout>
    </v-card-title>
    <v-divider />
    <v-card-text class="primary--text">
      <menu-item-list
        v-if="!newDialog"
        v-bind="$attrs"
        v-on="$listeners"
      />
      <slot
        v-else
        name="new-form"
        :close="() => newDialog = false"
      />
    </v-card-text>
    <v-divider />
  </v-card>
</template>

<script>
import MenuItemList from './MenuItemList'
export default {
  components: {
    MenuItemList
  },
  data () {
    return {
      newDialog: false
    }
  }

}
</script>
