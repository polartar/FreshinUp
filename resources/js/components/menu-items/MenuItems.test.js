import { mount } from '@vue/test-utils'

import * as Stories from './MenuItems.stories'

describe('components/menu-items/MenuItems', () => {
  describe('Snapshots', () => {
    test('Empty', async () => {
      const wrapper = mount(Stories.Empty())
      await wrapper.vm.$nextTick()
      expect(wrapper.element).toMatchSnapshot()
    })
    test('Populated', async () => {
      const wrapper = mount(Stories.Populated())
      await wrapper.vm.$nextTick()
      expect(wrapper.element).toMatchSnapshot()
    })
    test('Loading', async () => {
      const wrapper = mount(Stories.Loading())
      await wrapper.vm.$nextTick()
      expect(wrapper.element).toMatchSnapshot()
    })
  })
})
