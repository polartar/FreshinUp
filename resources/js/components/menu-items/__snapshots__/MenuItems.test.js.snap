// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/menu-items/MenuItems Empty Empty 1`] = `
<div
  class="v-card v-sheet theme--light"
  items=""
>
  <div
    class="v-card__title grey--text"
  >
    <div
      class="layout row space-between align-center"
    >
      <div
        class="flex"
      >
        <h3>
          Fleet Member Menu
        </h3>
      </div>
       
      <div
        class="flex shrink"
      >
        <button
          class="v-btn theme--light primary"
          slot="activator"
          text=""
          type="button"
        >
          <div
            class="v-btn__content"
          >
            <i
              aria-hidden="true"
              class="v-icon v-icon--left material-icons theme--dark"
            >
              add_circle_outline
            </i>
            Add new menu item
        
          </div>
        </button>
      </div>
    </div>
  </div>
   
  <hr
    class="v-divider theme--light"
  />
   
  <div
    class="v-card__text primary--text"
  >
    <div
      item-key="uuid"
    >
      <div
        class="ui-datatable elevation-1"
      >
        <div
          class="v-table__overflow"
        >
          <table
            class="v-datatable v-table v-datatable--select-all theme--light"
          >
            <thead>
              <tr>
                <th>
                  <div
                    class="v-input v-input--selection-controls v-input--checkbox v-input--hide-details theme--light"
                  >
                    <div
                      class="v-input__control"
                    >
                      <div
                        class="v-input__slot"
                      >
                        <div
                          class="v-input--selection-controls__input"
                        >
                          <input
                            aria-checked="false"
                            role="checkbox"
                            type="checkbox"
                            value=""
                          />
                          <div
                            class="v-input--selection-controls__ripple"
                          />
                          <i
                            aria-hidden="true"
                            class="v-icon material-icons theme--light"
                          >
                            check_box_outline_blank
                          </i>
                        </div>
                      </div>
                    </div>
                  </div>
                </th>
                <th
                  aria-label="Item: Not sorted. Activate to sort ascending."
                  aria-sort="none"
                  class="column sortable text-xs-left"
                  role="columnheader"
                  scope="col"
                  tabindex="0"
                >
                  <span>
                    
        Item
      
                  </span>
                  <i
                    aria-hidden="true"
                    class="v-icon material-icons theme--light"
                    style="font-size: 16px;"
                  >
                    arrow_upward
                  </i>
                </th>
                <th
                  aria-label="Cost: Not sorted. Activate to sort ascending."
                  aria-sort="none"
                  class="column sortable text-xs-right"
                  role="columnheader"
                  scope="col"
                  tabindex="0"
                >
                  <i
                    aria-hidden="true"
                    class="v-icon material-icons theme--light"
                    style="font-size: 16px;"
                  >
                    arrow_upward
                  </i>
                  <span>
                    
        Cost
      
                  </span>
                </th>
                <th
                  aria-label="Manage: Not sorted."
                  aria-sort="none"
                  class="column text-xs-center"
                  role="columnheader"
                  scope="col"
                >
                  <span>
                    
        Manage
      
                  </span>
                </th>
              </tr>
              <tr
                class="v-datatable__progress"
              >
                <th
                  class="column"
                  colspan="4"
                />
              </tr>
            </thead>
            <tbody>
              <tr>
                <td
                  class="text-xs-center"
                  colspan="4"
                >
                  No matching records found
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
       
      <div
        class="layout align-center"
      >
        <div
          class="flex grow justify-center"
        >
          <div
            class="layout justify-center"
          >
            <ul
              class="v-pagination theme--light"
            >
              <li>
                <button
                  class="v-pagination__navigation v-pagination__navigation--disabled"
                  type="button"
                >
                  <i
                    aria-hidden="true"
                    class="v-icon material-icons theme--light"
                  >
                    chevron_left
                  </i>
                </button>
              </li>
              <li>
                <button
                  class="v-pagination__item v-pagination__item--active primary"
                  type="button"
                >
                  1
                </button>
              </li>
              <li>
                <button
                  class="v-pagination__navigation v-pagination__navigation--disabled"
                  type="button"
                >
                  <i
                    aria-hidden="true"
                    class="v-icon material-icons theme--light"
                  >
                    chevron_right
                  </i>
                </button>
              </li>
            </ul>
          </div>
        </div>
         
        <div
          class="flex shrink"
        >
          <div
            class="v-input v-text-field v-select v-input--is-label-active v-input--is-dirty theme--light"
            role="combobox"
          >
            <div
              class="v-input__control"
            >
              <div
                class="v-input__slot"
              >
                <div
                  class="v-select__slot"
                >
                  <label
                    aria-hidden="true"
                    class="v-label v-label--active theme--light"
                    style="left: 0px; position: absolute;"
                  >
                    Results Per Page
                  </label>
                  <div
                    class="v-select__selections"
                  >
                    <div
                      class="v-select__selection v-select__selection--comma"
                    >
                      10
                    </div>
                    <input
                      aria-label="Results Per Page"
                      aria-readonly="false"
                      autocomplete="on"
                      readonly="readonly"
                      type="text"
                    />
                  </div>
                  <div
                    class="v-input__append-inner"
                  >
                    <div
                      class="v-input__icon v-input__icon--append"
                    >
                      <i
                        aria-hidden="true"
                        class="v-icon material-icons theme--light"
                      >
                        arrow_drop_down
                      </i>
                    </div>
                  </div>
                </div>
                <div
                  class="v-menu"
                >
                  <div
                    class="v-menu__content theme--light "
                    style="max-height: 300px; min-width: 0px; max-width: auto; top: 12px; left: 0px; transform-origin: top left; z-index: 0; display: none;"
                  >
                    <div
                      class="v-select-list v-card theme--light"
                    >
                      <div
                        class="v-list theme--light"
                        role="list"
                      >
                        <div
                          role="listitem"
                        >
                          <a
                            class="v-list__tile v-list__tile--link theme--light"
                          >
                            <div
                              class="v-list__tile__content"
                            >
                              <div
                                class="v-list__tile__title"
                              >
                                5
                              </div>
                            </div>
                          </a>
                        </div>
                        <div
                          class="primary--text"
                          role="listitem"
                        >
                          <a
                            class="v-list__tile v-list__tile--link v-list__tile--active theme--light primary--text"
                          >
                            <div
                              class="v-list__tile__content"
                            >
                              <div
                                class="v-list__tile__title"
                              >
                                10
                              </div>
                            </div>
                          </a>
                        </div>
                        <div
                          role="listitem"
                        >
                          <a
                            class="v-list__tile v-list__tile--link theme--light"
                          >
                            <div
                              class="v-list__tile__content"
                            >
                              <div
                                class="v-list__tile__title"
                              >
                                15
                              </div>
                            </div>
                          </a>
                        </div>
                        <div
                          role="listitem"
                        >
                          <a
                            class="v-list__tile v-list__tile--link theme--light"
                          >
                            <div
                              class="v-list__tile__content"
                            >
                              <div
                                class="v-list__tile__title"
                              >
                                25
                              </div>
                            </div>
                          </a>
                        </div>
                        <div
                          role="listitem"
                        >
                          <a
                            class="v-list__tile v-list__tile--link theme--light"
                          >
                            <div
                              class="v-list__tile__content"
                            >
                              <div
                                class="v-list__tile__title"
                              >
                                30
                              </div>
                            </div>
                          </a>
                        </div>
                        <div
                          role="listitem"
                        >
                          <a
                            class="v-list__tile v-list__tile--link theme--light"
                          >
                            <div
                              class="v-list__tile__content"
                            >
                              <div
                                class="v-list__tile__title"
                              >
                                50
                              </div>
                            </div>
                          </a>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="v-text-field__details"
              >
                <div
                  class="v-messages theme--light"
                >
                  <div
                    class="v-messages__wrapper"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
   
  <hr
    class="v-divider theme--light"
  />
</div>
`;

exports[`components/menu-items/MenuItems Empty Loading 1`] = `
<div
  class="v-card v-sheet theme--light"
  is-loading=""
>
  <div
    class="v-card__title grey--text"
  >
    <div
      class="layout row space-between align-center"
    >
      <div
        class="flex"
      >
        <h3>
          Fleet Member Menu
        </h3>
      </div>
       
      <div
        class="flex shrink"
      >
        <button
          class="v-btn theme--light primary"
          slot="activator"
          text=""
          type="button"
        >
          <div
            class="v-btn__content"
          >
            <i
              aria-hidden="true"
              class="v-icon v-icon--left material-icons theme--dark"
            >
              add_circle_outline
            </i>
            Add new menu item
        
          </div>
        </button>
      </div>
    </div>
  </div>
   
  <hr
    class="v-divider theme--light"
  />
   
  <div
    class="v-card__text primary--text"
  >
    <div
      item-key="uuid"
    >
      <div
        class="ui-datatable elevation-1"
      >
        <div
          class="v-table__overflow"
        >
          <table
            class="v-datatable v-table v-datatable--select-all theme--light"
          >
            <thead>
              <tr>
                <th>
                  <div
                    class="v-input v-input--selection-controls v-input--checkbox v-input--hide-details theme--light"
                  >
                    <div
                      class="v-input__control"
                    >
                      <div
                        class="v-input__slot"
                      >
                        <div
                          class="v-input--selection-controls__input"
                        >
                          <input
                            aria-checked="false"
                            role="checkbox"
                            type="checkbox"
                            value=""
                          />
                          <div
                            class="v-input--selection-controls__ripple"
                          />
                          <i
                            aria-hidden="true"
                            class="v-icon material-icons theme--light"
                          >
                            check_box_outline_blank
                          </i>
                        </div>
                      </div>
                    </div>
                  </div>
                </th>
                <th
                  aria-label="Item: Not sorted. Activate to sort ascending."
                  aria-sort="none"
                  class="column sortable text-xs-left"
                  role="columnheader"
                  scope="col"
                  tabindex="0"
                >
                  <span>
                    
        Item
      
                  </span>
                  <i
                    aria-hidden="true"
                    class="v-icon material-icons theme--light"
                    style="font-size: 16px;"
                  >
                    arrow_upward
                  </i>
                </th>
                <th
                  aria-label="Cost: Not sorted. Activate to sort ascending."
                  aria-sort="none"
                  class="column sortable text-xs-right"
                  role="columnheader"
                  scope="col"
                  tabindex="0"
                >
                  <i
                    aria-hidden="true"
                    class="v-icon material-icons theme--light"
                    style="font-size: 16px;"
                  >
                    arrow_upward
                  </i>
                  <span>
                    
        Cost
      
                  </span>
                </th>
                <th
                  aria-label="Manage: Not sorted."
                  aria-sort="none"
                  class="column text-xs-center"
                  role="columnheader"
                  scope="col"
                >
                  <span>
                    
        Manage
      
                  </span>
                </th>
              </tr>
              <tr
                class="v-datatable__progress"
              >
                <th
                  class="column"
                  colspan="4"
                >
                  <div
                    aria-valuemax="100"
                    aria-valuemin="0"
                    class="v-progress-linear"
                    role="progressbar"
                    style="height: 10px;"
                  >
                    <div
                      class="v-progress-linear__background primary"
                      style="height: 10px; opacity: 0.3; width: 100%;"
                    />
                    <div
                      class="v-progress-linear__bar"
                    >
                      <div
                        class="v-progress-linear__bar__indeterminate v-progress-linear__bar__indeterminate--active"
                      >
                        <div
                          class="v-progress-linear__bar__indeterminate long primary"
                        />
                        <div
                          class="v-progress-linear__bar__indeterminate short primary"
                        />
                      </div>
                      <!---->
                    </div>
                  </div>
                </th>
              </tr>
            </thead>
            <tbody>
              <tr>
                <td
                  class="text-xs-center"
                  colspan="4"
                >
                  No matching records found
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
       
      <div
        class="layout align-center"
      >
        <div
          class="flex grow justify-center"
        >
          <div
            class="layout justify-center"
          >
            <ul
              class="v-pagination v-pagination--disabled theme--light"
            >
              <li>
                <button
                  class="v-pagination__navigation v-pagination__navigation--disabled"
                  type="button"
                >
                  <i
                    aria-hidden="true"
                    class="v-icon material-icons theme--light"
                  >
                    chevron_left
                  </i>
                </button>
              </li>
              <li>
                <button
                  class="v-pagination__item v-pagination__item--active primary"
                  type="button"
                >
                  1
                </button>
              </li>
              <li>
                <button
                  class="v-pagination__navigation v-pagination__navigation--disabled"
                  type="button"
                >
                  <i
                    aria-hidden="true"
                    class="v-icon material-icons theme--light"
                  >
                    chevron_right
                  </i>
                </button>
              </li>
            </ul>
          </div>
        </div>
         
        <div
          class="flex shrink"
        >
          <div
            class="v-input v-text-field v-select v-input--is-label-active v-input--is-dirty theme--light"
            role="combobox"
          >
            <div
              class="v-input__control"
            >
              <div
                class="v-input__slot"
              >
                <div
                  class="v-select__slot"
                >
                  <label
                    aria-hidden="true"
                    class="v-label v-label--active theme--light"
                    style="left: 0px; position: absolute;"
                  >
                    Results Per Page
                  </label>
                  <div
                    class="v-select__selections"
                  >
                    <div
                      class="v-select__selection v-select__selection--comma"
                    >
                      10
                    </div>
                    <input
                      aria-label="Results Per Page"
                      aria-readonly="false"
                      autocomplete="on"
                      readonly="readonly"
                      type="text"
                    />
                  </div>
                  <div
                    class="v-input__append-inner"
                  >
                    <div
                      class="v-input__icon v-input__icon--append"
                    >
                      <i
                        aria-hidden="true"
                        class="v-icon material-icons theme--light"
                      >
                        arrow_drop_down
                      </i>
                    </div>
                  </div>
                </div>
                <div
                  class="v-menu"
                >
                  <div
                    class="v-menu__content theme--light "
                    style="max-height: 300px; min-width: 0px; max-width: auto; top: 12px; left: 0px; transform-origin: top left; z-index: 0; display: none;"
                  >
                    <div
                      class="v-select-list v-card theme--light"
                    >
                      <div
                        class="v-list theme--light"
                        role="list"
                      >
                        <div
                          role="listitem"
                        >
                          <a
                            class="v-list__tile v-list__tile--link theme--light"
                          >
                            <div
                              class="v-list__tile__content"
                            >
                              <div
                                class="v-list__tile__title"
                              >
                                5
                              </div>
                            </div>
                          </a>
                        </div>
                        <div
                          class="primary--text"
                          role="listitem"
                        >
                          <a
                            class="v-list__tile v-list__tile--link v-list__tile--active theme--light primary--text"
                          >
                            <div
                              class="v-list__tile__content"
                            >
                              <div
                                class="v-list__tile__title"
                              >
                                10
                              </div>
                            </div>
                          </a>
                        </div>
                        <div
                          role="listitem"
                        >
                          <a
                            class="v-list__tile v-list__tile--link theme--light"
                          >
                            <div
                              class="v-list__tile__content"
                            >
                              <div
                                class="v-list__tile__title"
                              >
                                15
                              </div>
                            </div>
                          </a>
                        </div>
                        <div
                          role="listitem"
                        >
                          <a
                            class="v-list__tile v-list__tile--link theme--light"
                          >
                            <div
                              class="v-list__tile__content"
                            >
                              <div
                                class="v-list__tile__title"
                              >
                                25
                              </div>
                            </div>
                          </a>
                        </div>
                        <div
                          role="listitem"
                        >
                          <a
                            class="v-list__tile v-list__tile--link theme--light"
                          >
                            <div
                              class="v-list__tile__content"
                            >
                              <div
                                class="v-list__tile__title"
                              >
                                30
                              </div>
                            </div>
                          </a>
                        </div>
                        <div
                          role="listitem"
                        >
                          <a
                            class="v-list__tile v-list__tile--link theme--light"
                          >
                            <div
                              class="v-list__tile__content"
                            >
                              <div
                                class="v-list__tile__title"
                              >
                                50
                              </div>
                            </div>
                          </a>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="v-text-field__details"
              >
                <div
                  class="v-messages theme--light"
                >
                  <div
                    class="v-messages__wrapper"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
   
  <hr
    class="v-divider theme--light"
  />
</div>
`;
